common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryAuthorizationResultFactory.java

```diff
@@ -0,0 +1,132 @@
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationResult;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationResultFactory;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationStatus;
+import com.microsoft.identity.common.internal.util.StringUtil;
+
+import java.io.Serializable;
+import java.util.HashMap;
+
+public class AzureActiveDirectoryAuthorizationResultFactory extends AuthorizationResultFactory {
+
+    private static final String TAG = AzureActiveDirectoryAuthorizationResultFactory.class.getSimpleName();
+
+    private static final String ERROR_CODES = "error_codes";
+
+    @Override
+    public AuthorizationResult createAuthorizationResult(final int resultCode, final Intent data) {
+        if (data == null || data.getExtras() == null) {
+            return createAuthorizationResultWithErrorResponse(AuthorizationStatus.FAIL,
+                    MicrosoftAuthorizationErrorResponse.AUTHORIZATION_FAILED,
+                    MicrosoftAuthorizationErrorResponse.NULL_INTENT);
+        }
+        final Bundle extras = data.getExtras();
+        final int requestId = extras.getInt(AuthenticationConstants.Browser.REQUEST_ID);
+        switch (resultCode) {
+            case AuthenticationConstants.UIResponse.BROWSER_CODE_CANCEL:
+                Logger.verbose(TAG, "User cancel the request in webview: " + requestId);
+                return createAuthorizationResultWithErrorResponse(AuthorizationStatus.USER_CANCEL,
+                        MicrosoftAuthorizationErrorResponse.USER_CANCEL,
+                        MicrosoftAuthorizationErrorResponse.USER_CANCELLED_FLOW);
+
+            case AuthenticationConstants.UIResponse.BROWSER_CODE_COMPLETE:
+                final String url = extras.getString(AuthenticationConstants.Browser.RESPONSE_FINAL_URL, "");
+                return parseUrlAndCreateAuthorizationResult(url);
+
+            case AuthenticationConstants.UIResponse.BROWSER_CODE_ERROR:
+                // This is purely client side error, possible return could be chrome_not_installed or the request intent is
+                // not resolvable
+                final String error = extras.getString(AuthenticationConstants.Browser.RESPONSE_ERROR_CODE);
+                final String errorDescription = extras.getString(AuthenticationConstants.Browser.RESPONSE_ERROR_MESSAGE);
+                return createAuthorizationResultWithErrorResponse(AuthorizationStatus.FAIL, error, errorDescription);
+
+            case AuthenticationConstants.UIResponse.BROWSER_CODE_AUTHENTICATION_EXCEPTION:
+                //TODO : Verify that a ClientException is serialized here after Broker Implementation
+                Serializable clientException = extras.getSerializable(AuthenticationConstants.Browser.RESPONSE_AUTHENTICATION_EXCEPTION);
+                if (clientException != null && clientException instanceof ClientException) {
```

Comment:
`clientException` seems an assumptive name if we have to test
`(clientException instanceof ClientException)`

Let's call it something like `responseAuthenticationException` before we cast it. After the cast, this name seems fine

Reply:
Done.